csi.proto

```diff
@@ -301,6 +313,203 @@ message Volume {
+  // "us-west1" and the "zone" "us-west1-b".
+  //
+  // Example 2:
+  // Given a volume that is accessible from a single zone, and
+  // accepted =
+  //   { "region": {"us-west1"}, "zone": {"us-west1-b", "us-west1-c"}}
+  // Indicates the volume may be provisioned in either the "zone"
+  // "us-west1-b" or the "zone" "us-west1-c" in the "region" "us-west1".
+  //
+  // Example 3:
+  // Given a volume that is accessible from two zones (which may be
+  // indicated via an opaque parameter if the SP supports both single
+  // and multi-zone volumes), and
+  // accepted =
+  //   {"region": {"us-west1"}, "zone": {"us-west1-b", "us-west1-c"}}
+  // Indicates the volume MUST be accessible from the "region"
+  // "us-west1" and both "zone" "us-west1-b" and "zone" "us-west1-c".
+  //
+  // Example 4:
+  // Given a volume that is accessible from two zones (which may be
+  // indicated via an opaque parameter if the SP supports both single
+  // and multi-zone volumes), and
+  // accepted =
+  //   {"region": {"us-west1"},
+  //    "zone": {"us-west1-b", "us-west1-c", "us-west1-d"}}
+  // Indicates the volume may be accessible from any combination of two
+  // "zones" in the "region" "us-west1": e.g. "us-west1-b" and
+  // "us-west1-c" or "us-west1-b" and "us-west1-d", etc.
+  //
+  // Example 5:
+  // Given a volume that is accessible from a single site/zone/rack, and
+  // accepted =
+  //   {"site": {"S1"}, "zone": {"Z1"}, "rack": {"R1"}}
+  // Indicates the volume MUST be accessible from the "site" "S1", the
+  // the "zone" "Z1", and the "rack" "R1".
+  //
+  // Example 6:
+  // Given a volume that is accessible from a single site/zone/rack, and
+  // accepted =
+  //   {"site": {"S1"}, "zone": {"Z1"}, "rack": {"R1", "R2"}}
+  // Indicates the volume MUST be accessible from either "rack" "R1" or
+  // "R2" in "zone" "Z1" in "site" "S1".
+  //
+  // Example 7:
+  // Given a volume that is accessible from a single site/zone/rack, and
+  // accepted =
+  //   {"site": {"S1"}, "zone": {"Z1", "Z2"}, "rack": {"R1", "R2"}}
+  // Indicates the volume MUST be accessible from either "rack" "R1" or
+  // "R2" in either "zone" "Z1" or "Z2" in the "site" "S1".
+  map<string, ListOfString> accepted = 1;
+
+  // If a CO specifies multiple values for a topological segment, it may
+  // also specify one or more preferred values.  An SP SHOULD attempt
+  // to fullfil the provisioning request using the preferred value(s) in
+  // order from first to last and only use the the other
+  // accepted topologies values if it is not possible to fulfill the
+  // CreateVolume request with the preferred value(s).
+  // Each key MUST also be present in accepted topologies.
+  // Each value MUST be present in the list of values for the
+  // corresponding key in accepted topologies.
+  // This field is OPTIONAL. If it is not specified, the SP MAY choose
+  // from any of the set of possible values specified in
+  // accepted topologies.
```

Comment:
Could clarify that while the CO specifies a _preferred_ value(s), the SP should be able to make the volume _accessible from_ values in the preferred and the accepted list. Or is that not a requirement? In which case, once the volume is created, the CO can't expect the volume to be accessible outside of the topology segment in the preferred list.

Reply:
Any values in the preferred map must exist in the accepted list. The accepted list defines the list of possibilities. The optional preferred list simply specifies which of those are preferred by the CO. The SP must not provision outside of what is dictated in the accepted list.